Server/States/RunningState.cpp

```diff
@@ -421,8 +422,51 @@ void RunningState::UpdatePlayerMovement(GameObject* player, const InputPacket& i
         networkData->outgoingFunctions.Push(std::make_pair(id, FunctionPacket( Replicated::Grapple_Event , &data)));
         playerMovement->cameraAnimationCalls.grapplingEvent = 0;
     }
+
+    if (playerMovement->uiAnimationData.grapplingAvailability != -1) {
+        auto id = GetIdFromPlayerObject(player);
+        FunctionData data;
+        DataHandler handler(&data);
+        handler.Pack(playerMovement->uiAnimationData.grapplingAvailability);
+        networkData->outgoingFunctions.Push(std::make_pair(id, FunctionPacket(Replicated::GameInfo_GrappleAvailable, &data)));
+        playerMovement->uiAnimationData.grapplingAvailability = -1;
+    }
+}
+
+void RunningState::UpdatePlayerGameInfo(GameObject* player, const InputPacket& inputInfo) {
+    UpdateGameTimerInfo(player, inputInfo);
+    UpdatePlayerPositionsInfo(player, inputInfo);
+}
+
+void RunningState::UpdateGameTimerInfo(GameObject* player, const InputPacket& inputInfo) {
+    auto id = GetIdFromPlayerObject(player);
+    FunctionData data;
+    DataHandler handler(&data);
+    handler.Pack(levelManager->GetElapsedTime());
+    handler.Pack(levelManager->GetPlatinumTime());
+    handler.Pack(levelManager->GetGoldTime());
+    handler.Pack(levelManager->GetSilverTime());
+
+    int medalID = (int)levelManager->GetCurrentMedal();
+    handler.Pack(medalID);
+    networkData->outgoingFunctions.Push(std::make_pair(id, FunctionPacket(Replicated::GameInfo_Timer, &data)));
 }
 
+void RunningState::UpdatePlayerPositionsInfo(GameObject* player, const InputPacket& inputInfo) {
```

Comment:
This is okay but would have been better to just get the positions client side instead of sending positions again